doc/04-Upgrading.md

```diff
@@ -3,6 +3,64 @@
+analysis on another console and unblock it if necessary. It is important to note though that the script may need some
+time to perform the reindexing on the `history` table even if it is not blocked. Nonetheless, you can use the
+`show processlist` command to determine whether an excessive number of queries have been stuck in a waiting state.
+
+```
+MariaDB [icingadb]> show processlist;
++------+-----+-----+----------+-----+------+---------------------------------+------------------------------------+-----+
+| Id   | ... | ... | db       | ... | Time | State                           | Info                               | ... |
++------+-----+-----+----------+-----+------+---------------------------------+------------------------------------+-----+
+| 1475 | ... | ... | icingadb | ... | 1222 | Waiting for table metadata lock | INSERT INTO "notification_history" | ... |
+| 1485 | ... | ... | icingadb | ... | 1262 | Creating sort index             | SELECT history.id, history....     | ... |
+| 1494 | ... | ... | icingadb | ... | 1224 | Waiting for table metadata lock | ALTER TABLE history ADD INDEX ...  | ... |
+| 1499 | ... | ... | icingadb | ... | 1215 | Waiting for table metadata lock | INSERT INTO "notification_history" | ... |
+| 1500 | ... | ... | icingadb | ... | 1215 | Waiting for table metadata lock | INSERT INTO "state_history" ...    | ... |
++------+-----+-----+----------+-----+------+---------------------------------+------------------------------------+-----+
+39 rows in set (0.001 sec)
+```
+
+In the above output you can see that there are approximately `39` queries (`39 rows in set (0.001 sec)`), including the
+`ALTER TABLE history ADD INDEX` query from the upgrade script, waiting for a metadata lock, they're just minimised to
+the bare essentials. Unfortunately, only one of these queries is holding the `table metadata lock` that everyone else
+is now waiting for, which in this case is a `SELECT` statement initiated by Icinga DB Web in the `icingadb/history`
+view, which takes an unimaginably long time. Note that there might be multiple `SELECT` statements started before the
+upgrade script in your case when the history view of your Icinga DB Web is opened by different Icinga Web users at the
+same time.
+
+You can now either just wait for the `SELECT` statements to finish by themselves and let them block the upgrade script
+and all Icinga DB queries on all `*_history` tables or forcibly terminate them and let the remaining queries do their
+work. In this case, cancelling that one blocking `SELECT` query will let the upgrade script continue normally without
+blocking any other queries.
+``` 
+MariaDB [icingadb]> kill 1485;
+```
+
+In case you are insecure about which Icinga DB Web queries are blocking, you may simply cancel all `SELECT` statements
+listed with `show processlist`. Cancelling a `SELECT` query is usually not that disastrous, so feel free to abort every
+single one of them.
```

Comment:
I'd be more cautious with such suggestions. Keep in mind that `show processlist` shows everything on the server including other databases/applications if present. I wouldn't want to recommend "just kill it" for everything, maybe just for long-running (see time column) selects on that specific database.